lib/src/index.ts

```diff
@@ -368,13 +371,26 @@ export class NanoTDFDatasetClient extends Client {
 
     const version = '0.0.1';
     // Rewrap key on every request
+    if (this.isRewrapFailed) {
```

Comment:
Are we able to inspect `nanoTdfHeader`?  If we can add an inexpensive check to see if the Policy changed.  So throttle if the same policy issued that got a 403.  If the policy is different then hit rewrap endpoint.

Reply:
Updated